File: Backend/WebMedicalApplication/BackendMedicalApplication/DTo/AppointmentDto.cs
namespace BackendMedicalApplication.DTo
{
    public class AppointmentDto
    {
        public int AppointmentId { get; set; }
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public DateTime DateTime { get; set; }
        public string Reason { get; set; }
        public int? PatientRecordId { get; set; }
        public PatientRecordDto? PatientRecord { get; set; }
    }

}

File: Frontend/FrontMedicalApplication/src/app/core/dtos/appointment.dto.ts
export class AppointmentDto {
  appointmentId?: number;
  patientId: number;
  doctorId: number;
  dateTime: Date;
  reason?: string;
  patientRecordId?: number;
  status?: string;

  constructor(
    patientId: number,
    doctorId: number,
    dateTime: Date = new Date(),
    reason?: string,
    appointmentId?: number,
    patientRecordId?: number,
    status?: string
  ) {
    this.patientId = patientId;
    this.doctorId = doctorId;
    this.dateTime = dateTime;
    this.reason = reason;
    this.status = status;
    if (appointmentId) {
      this.appointmentId = appointmentId;
    }
    if (patientRecordId) {
      this.patientRecordId = patientRecordId;
    }
  }
}
